File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.scala
object Solution {
  def minSubArrayLen(target: Int, nums: Array[Int]): Int = {
    var total = 0
    var left = 0
    var right = 0
    var res = Int.MaxValue
    val n = nums.length

    while (left <= right && right < n) {
      total += nums(right)
      while (total >= target) {
        res = res.min(right - left + 1)
        total -= nums(left)
        left += 1
      }
      right += 1
    }

    if (res != Int.MaxValue) res else 0
  }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int total =0, left =0, right=0;
        int res = Integer.MAX_VALUE;
        int n = nums.length;

        while(left<=right && right < n){

            total += nums[right];
            while(total>=target){

                res = Math.min(res, right-left + 1);
                total -= nums[left++];
            }
            right++;
        
        }
        if(res != Integer.MAX_VALUE) 
            return res;
        else
            return 0;
    }
}